### PR TITLE
Fixed handling of relationship direction in ExpandIntoPipe

### PR DESCRIPTION
**Pull request by lutovich**
`ExpandIntoPipe` contains a cache of relationships which is very valuable for cyclic patterns.
This cache was not aware about direction and could serve wrong relationships for pair of nodes.
This commit makes cache use relationship directions and cleans up cache implementation.

### CODE REVIEW

**Comment by davidegrohmann** [comment]
@lutovich What about having some tests with `Direction.BOTH`?

**Comment by lutovich** [comment]
@davidegrohmann missed this. Will verify

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/pipes/ExpandIntoPipe.scala [hunk 1]
```diff
@@ -148,19 +151,29 @@ case class ExpandIntoPipe(source: Pipe,
...
+      put(start, end, RelationshipsCache.NoRels)
+      put(end, start, RelationshipsCache.NoRels)
+    }
+
+    def get(start: Node, end: Node) = table.get(key(start, end))
 
     @inline
-    private def key(node1: Node, node2: Node) =
-      if (node1.getId < node2.getId) (node1, node2) else (node2, node1)
+    private def key(start: Node, end: Node) = (start.getId, end.getId)
```
Comment (jexp) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/pipes/ExpandIntoPipe.scala]: is that the fix? Or why would you add the tuple potentially twice?

Comment (lutovich) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/pipes/ExpandIntoPipe.scala]: Changed code a bit. For `Direction.BOTH` ordering in tuple is not important so we sort elements. For `Direction.OUTGOING` and `Direction.INCOMING` ordering is important - cache should serve different sets of relationships for `(Node1, Node2)` and `(Node2, Node1)`.

**Comment by lutovich** [comment]
@davidegrohmann case with `Direction.BOTH` is addressed now

**Pull request merged**